components/display/ili9341.rst

```diff
@@ -0,0 +1,123 @@
+ILI9341 TFT LCD
+===============
+
+.. seo::
+    :description: Instructions for setting up ILI9341 TFT LCD display drivers.
+    :image: 24-ili9341-full.jpg
+
+.. _ili9341:
+
+Usage
+-----
+
+The ``ili9341`` display platform allows you to use
+ILI9341 (`datasheet <https://cdn-shop.adafruit.com/datasheets/ILI9341.pdf>`__,
+`Ali model used <https://s.click.aliexpress.com/e/_dTKH6Mt>`__)
+displays with ESPHome. Note that this display controller uses a lot of pins even if touch
+is not used. So it's better to used it with an EPS32 though esp8266 might be suitable
+too (with some SPI flash changes). 
+
+.. figure:: images/24-ili9341-full.jpg
+    :align: center
+    :width: 75.0%
+
+    ILI9341 common display
+
+
+.. code-block:: yaml
+
+    # Example minimal configuration entry
+    display:
+      - platform: ili9341
+        model: TFT_2.4
+        cs_pin: 14
+        dc_pin: 27
+        # led_pin: 32 // you may use it with a constant brightness but there is a better way
+        reset_pin: 33
+        rotation: 0
+        
+        lambda: |-
+          it.fill(ili9341::BLACK);
+          it.print(0, 0, id(font), "Hello World!");
+    
+Configuration variables
+***********************
+
+- **led_pin** (:ref:`Pin Schema <config-pin_schema>`): The display's backlight pin.
+- **cs_pin** (:ref:`Pin Schema <config-pin_schema>`): The CS pin.
+- **dc_pin** (:ref:`Pin Schema <config-pin_schema>`): The DC pin.
```

Comment:
those mark required, please sort them by required then optional
also add ``model`` required

Reply:
Hey @glmnet what do you mean by add a model?

Reply:
there is a model property!, marked required see here https://github.com/esphome/esphome/blob/d20caa9d60e91b3044467f16eedfef9ab89d1299/esphome/components/ili9341/display.py#L29